server/config/config.go

```diff
@@ -39,6 +38,24 @@ type StoredConfig struct {
 	// added, and the Manifest struct is stored in KV under
 	// manifest_<sha1(Manifest)>. Implementation in `store.Manifest`.
 	LocalManifests map[string]string `json:"local_manifests,omitempty"`
+
+	DeveloperMode bool `json:"developer_mode"`
+	AllowHTTPApps bool `json:"allow_http_apps"`
+}
+
+func unmarshalPluginConfig(pluginConfig map[string]any, mmconf *model.Config, mattermostCloudMode bool) StoredConfig {
+	sc := StoredConfig{}
+	utils.Remarshal(sc, pluginConfig)
+
+	if _, ok := pluginConfig["developer_mode"]; !ok {
+		sc.DeveloperMode = pluginapi.IsConfiguredForDevelopment(mmconf)
+	}
+
+	if _, ok := pluginConfig["allow_http_apps"]; !ok {
+		sc.AllowHTTPApps = sc.DeveloperMode || !mattermostCloudMode
```

Comment:
Shouldn't this be and `&&`? If developermode is on and the mattermostcloud is not active?

Reply:
The logic works as intended. For on-prem servers we want to allow HTTP apps by default.